src/components/Table.tsx

```diff
@@ -0,0 +1,58 @@
+import * as React from 'react';
+import {cn} from "../utils/util";
+
+
+export interface TableColumn {
+    field: string;
+    headerName: string;
+    width: number;
+    className?: string;
+}
+
+export interface TableRow {
+    [key: string]: React.ReactNode;
+}
+
+interface CardProps {
+    rows: TableRow[];
+    columns: TableColumn[];
+    className?: string;
+}
+
+export function Table({rows, columns, className}: CardProps) {
```

Comment:
Currently, your table component does not check if the keys of the objects in the rows prop match the field names in the columns prop. This means that consumers of this component can pass in rows that have missing or extra keys.

There might be a way to do this using typescript's type system. I think.
Can you please look into this?

Reply:
I couldn't find a way to do this unfortunately